chat-plugins/datasearch.js

```diff
@@ -1382,7 +1382,8 @@ function runItemsearch(target, cmd, canAll, message) {
 		}
 
 		// iterate over found items again to make sure they all are the best match
-		for (let [i, id] of foundItems.entries()) {
+		// use a reversed copy for iteration so we can remove items mid-loop without changing the remaining indices
+		for (let [i, id] of foundItems.map((element, index) => [index, element]).reverse()) {
```

Comment:
Also `const`, which would have found you the bug where you're pointlessly doing `i--`